Update nf-azure: Add Azure Batch node communication mode option (#7337)
nextflow-io/nextflow: docs/reference/config.mdx

The `startTask` that is executed as the node joins the Azure Batch node pool.

##### `azure.batch.pools.<name>.targetCommunicationMode`

> The node communication mode used by the pool. Can be either `classic` or `simplified`. When not set, the Azure Batch service selects the mode. In `simplified` mode the Batch service initiates communication with the compute nodes, reducing the outbound network access required by the nodes.

Reviewer: ```suggestion
The node communication mode used by the pool. One of `classic` or `simplified`. When not set, the Azure Batch service selects the mode. In `simplified` mode, the Batch service initiates communication with the compute nodes, which reduces the outbound network access the nodes require.

```
